src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -119,26 +132,43 @@ def __init__(
             asyncio.CancelledError: when the task gets cancelled.
         """
         assert (
-            sampling_period.total_seconds() > 0
-        ), "The sampling period should be greater than zero."
+            input_sampling_period.total_seconds() > 0
+        ), "The input sampling period should be greater than zero."
         assert (
-            sampling_period <= size
-        ), "The sampling period should be equal to or lower than the window size."
+            input_sampling_period <= size
+        ), "The input sampling period should be equal to or lower than the window size."
+
+        sampling = input_sampling_period
+        self._resampler: Resampler | None = None
+        self._avg_sender: Sender[Sample] | None = None
+        self._resampler_task: asyncio.Task[None] | None = None
+
+        if resampler_config:
+            resampling_period = timedelta(seconds=resampler_config.resampling_period_s)
+            assert (
+                resampling_period <= size
+            ), "The resampling period should be equal to or lower than the window size."
+
+            if resampling_period != input_sampling_period:
+                self._resampler = Resampler(resampler_config)
+                sampling = resampling_period
 
         # Sampling period might not fit perfectly into the window size.
-        num_samples = math.ceil(size / sampling_period)
+        num_samples = math.ceil(size.total_seconds() / sampling.total_seconds())
```

Comment:
I know it is very simple code, but I've seen this before. Maybe move it to a `_get_num_samples()` utility method to make sure the way is calculated is consistent everywhere we calculate it?

Reply:
This is a special case and it is only used to set the size of the underlying ring buffer when it is instantiated. A user can get the size of the moving window by calling len(moving_window) as `__len__` was implemented (recently fixed in #274)